lms/static/coffee/src/instructor_dashboard/queries.coffee

```diff
@@ -0,0 +1,679 @@
+      curRow.remove()
+      if curRow.hasAttribute('query')
+        queryToDelete = curRow.getAttribute('query')
+        @delete_temp_query(queryToDelete)
+      @check_done()
+    $td = $ '<td>'
+    $td.append($removeBtn)
+    row.appendChild($td[0])
+    return $(row)
+
+  save_query: (cb)->
+    cur_queries = []
+    tab = $('.emailWidget.queryTableBody')
+    rows = tab.find('tr')
+    _.each rows, (row) ->
+      cur_queries.push(row.getAttribute('query'))
+    send_data =
+      existing: cur_queries.join(',')
+      savedName: $(".emailWidget.savequeryname").val()
+    $(".emailWidget.savequeryname").val("")
+    $.ajax(
+      type: 'POST'
+      dataType: 'json'
+      url: @$saveQueryBtn.data('endpoint')
+      data: send_data
+      success: (data) -> cb? null, data
+      error: std_ajax_err ->
+        cb? gettext('Error saving query')
+    )
+  # save queries in active queries
+  send_save_query: ->
+    @save_query (error, data) =>
+      if error
+        return @show_errors error
+      $('#send_email select').append('<option value="' + data['group_id'] + '">' + data['group_title'] + '</option>')
+      @load_saved_queries()
+
+  get_estimated: (cb)->
+    curQueries = []
+    tab = $('.emailWidget.queryTableBody')
+    rows = tab.find('tr')
+    _.each rows, (row) ->
+      curQueries.push(row.getAttribute('query'))
+    send_data =
+      existing: curQueries.join(',')
+    $.ajax
+      dataType: 'json'
+      url: @$totalEndpoint
+      data: send_data
+      success: (data) -> cb? null, data
+      error: std_ajax_err ->
+        cb? gettext('Error getting estimated')
+
+  # estimate the students selected
+  reload_estimated: ->
+    $('.emailWidget.estimated').html(gettext('Calculating'))
+    @get_estimated (error, students) =>
+      if students['success'] == false
+        $('.emailWidget.estimated').html(gettext('0 students selected'))
+        return
+      studentsList = students['data']
+      queryId = students['query_id']
+      # abort on error
+      if error
+        return @show_errors error
+      $numberStudents = studentsList.length
+      $('.emailWidget.estimated').html(gettext('approx ' + $numberStudents + ' students selected'))
+  # set error display
+  show_errors: (msg) -> @$error_section?.text msg
+
+#Queries Section
+class Queries
+  # enable subsections.
+  constructor: (@$section) ->
+    # attach self to html
+    # so that instructor_dashboard.coffee can find this object
+    # to call event handlers like 'onClickTitle'
+    @$section.data 'wrapper', @
+
+    #initialize email widget selectors
+    @$emailListContainers = @$section.find('.email-list-container')
+    @emailLists = _.map (@$emailListContainers), (email_list_container) =>
+      new EmailSelectors $(email_list_container), @$section
+
+    #initialize email widget
+    new EmailWidget(@emailLists, @$section, @$emailListContainers)
+    # populate selector
+
+  # handler for when the section title is clicked.
+  onClickTitle: ->
```

Comment:
It would probably be good to follow something like here https://github.com/Stanford-Online/edx-platform/blob/master/lms/static/coffee/src/instructor_dashboard/data_download.coffee#L133

so when you come to this tab, error messages and stuff from before are cleared. And you only call `poller.start` to purge temporary queries if you are actually on this tab.

likewise adding an `onExit` to stop the poller when you leave this tab.